pkg/oci/client/compute.go

```diff
@@ -93,13 +93,23 @@ func (c *client) listVNICAttachments(ctx context.Context, req core.ListVnicAttac
 
 func (c *client) GetPrimaryVNICForInstance(ctx context.Context, compartmentID, instanceID string) (*core.Vnic, error) {
 	var page *string
+
+	if compartmentID == "" {
+		glog.V(6).Infof("No compartment specified, looking up instance: %s", instanceID)
```

Comment:
We always know the compartment don't we? We make that assumption else where in the code since we fetch it from metadata or the configuration.